requests_toolbelt/adapters/x509_adapter.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+"""A X509Adapter for use with the requests library.
+
+This file contains an implementation of the X509Adapter that will
+allow users to authenticate a request using an arbitrary
+X.509 certificate without needing to convert it to a .pem file
+
+"""
+
+from OpenSSL.crypto import PKey, X509
+from cryptography import x509
+from cryptography.hazmat.primitives.serialization import (load_pem_private_key,
+                                                          load_der_private_key)
+from cryptography.hazmat.primitives.serialization import Encoding
+from cryptography.hazmat.backends import default_backend
+
+from datetime import datetime
+from requests.adapters import HTTPAdapter
+import requests
+
+from .._compat import PyOpenSSLContext
+from .. import exceptions as exc
+
+try:
+    from _ssl import PROTOCOL_TLS as PROTOCOL
```

Comment:
i used `_ssl` because i didn't need any of the actual implementation portions of ssl, just some of the constants that are defined in `_ssl` and passed through `ssl`.  It also solved an issue i was having trying to import directly from `ssl` that only occurred in the tox py27 environment.

Reply:
I'm surprised that you ran into issues with importing `ssl` on Py27. Would you be willing to add a comment explaining that? I just don't want to come back to this in the future, change the import and break something for someone.

What version of Python 2.7 are you using by the way?

Reply:
I'll add a comment explaining that.  I'm using 2.7.15, but it actually happened under tox in 2.7.14.  I tried several things to try to correct it, but the switch to using _ssl was the only thing that worked on the 2.7.x and 3.x lines.